src/segment_set.rs

```diff
@@ -428,6 +434,7 @@ impl SegmentSet {
                 }
             }
             for seg in segments {
+                #[allow(clippy::branches_sharing_code)]
```

Comment:
Suggestion:
```rust
let has_next = seg.0.next_file != Span::null();
state.segments.push(seg);
if has_next { ... }
```
Alternatively (more tersely):
```rust
if (seg.0.next_file != Span::null(), state.segments.push(seg)).0 { ... }
```
using the `(x, y).0` idiom for executing `y` after `x` but returning `x`.